### DISCUSSION TITLE
Contract read not working (Seemingly on wrong network)

### DISCUSSION THREAD

**Discussion by L-TheG** [wevm/wagmi#425] [Q&A]
Hi,
I am trying to use the useContractRead hook.
When I am loading the page that uses the hook I get the error '_contract.interface.encodeFunctionData is not a function_' in the browser console and no value from the hook.

I am setting the hook up like this:
```
import ContractAbi from '../artifacts/contracts/Contract.sol/Contract.json';

function Component() {
    const contractAbi = new ethers.utils.Interface(ContractAbi.abi);
    const {data, isError, isLoading, refetch} = useContractRead(
        {
            addressOrName: contractAddress,
            contractInterface: contractAbi,
        },
        'ctrBalance'
    );
}
```
The function I am calling is this (confirmed to work through testing):
```
// Contract.sol
function ctrBalance() public view onlyOwner returns (uint256) {
    return address(this).balance;
}
```
This is all running on a local hardhat environment.
All useContractWrite hooks in the same app work as intended.
Does anyone see an issue with my setup?

Thanks :)

**Comment by L-TheG** [comment]
I dont want to open another discussion as this seems related.

So my setup for the provider and connector in Next.js is as follows:
```
const client = createClient({
    autoConnect: true,
   // The hardhat chainId is manually set to 1337 in the hardhatConfig file
    provider: new JsonRpcProvider('http://127.0.0.1:8545/', getNetwork(1337)),
    connectors: [new MetaMaskConnector({chains: [chain.localhost]})],
});

function MyApp({Component, pageProps}) {
    <WagmiProvider client={client}>
          <Component {...pageProps}></Component>
    </WagmiProvider>
}
```
The component using the contract read hook, now looks like this:
```
import ContractAbi from '../artifacts/contracts/Contract.sol/Contract.json';

function Component() {
    const provider = useProvider();
    const {refetch} = useContractRead(
        {
            addressOrName: contractAddress,
            contractInterface: ContractAbi.abi,
        },
        'ctrBalance'
    );

    const handleButtonPress = async () => {
        console.log(await refetch());
        console.log(prov._network);
    }
}

```
The result of await refetch is error with message: 'failed to meet quorum' with a lot more specific info (don't think its needed here)
The result of prov._network is that I am on homestead with chainId 1. So mainnet.

I am at a loss of what to do. I don't really know where to go from here since I can't seem to get the Provider to use the localhost-hardhat network to test my App.

Does anyone have some pointers on how to properly set up my provider (since I am suspecting that something is wrong with it).

**Edit**: Without changing anything I am now getting: 
'call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="ctrBalance()", data="0x", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.6.1)'
as the return value of 'console.log(await refetch());'

Thanks!

**Reply by thecooltechguy** [reply]
I'm facing this same issue and I'm also using Next.js. The CALL_EXCEPTION error that's thrown, in this case, is simply because the library is trying to read from the address on mainnet, instead of the local chain.

**Comment by dobbobalina2** [comment]
> Hi, I am trying to use the useContractRead hook. When I am loading the page that uses the hook I get the error '_contract.interface.encodeFunctionData is not a function_' in the browser console and no value from the hook.
> 
> I am setting the hook up like this:
> 
> ```
> import ContractAbi from '../artifacts/contracts/Contract.sol/Contract.json';
> 
> function Component() {
>     const contractAbi = new ethers.utils.Interface(ContractAbi.abi);
>     const {data, isError, isLoading, refetch} = useContractRead(
>         {
>             addressOrName: contractAddress,
>             contractInterface: contractAbi,
>         },
>         'ctrBalance'
>     );
> }
> ```
> 
> The function I am calling is this (confirmed to work through testing):
> 
> ```
> // Contract.sol
> function ctrBalance() public view onlyOwner returns (uint256) {
>     return address(this).balance;
> }
> ```
> 
> This is all running on a local hardhat environment. All useContractWrite hooks in the same app work as intended. Does anyone see an issue with my setup?
> 
> Thanks :)

yea I'm getting a similar issue now. Not quite sure why.

**Comment by dobbobalina2** [comment]
For me it was two issues.

1.  I woukdn't get the abi like that. Import it directly so instead of`  contract.abi` just  `abi `
2. if you put `enabled = false` in the configs for the call that will get rid of the error.

**Comment by brucdarc** [comment]
I also seem to be having this issue.

It seems when I call the hook by default, the status is always idle. I'm not sure why this is, I'm able to make contract calls successfully using the user's wallet just fine.
